handler/schedule.go

```diff
@@ -125,3 +137,65 @@ func (s *ScheduleHandler) mergeResourcesWithDefaults(resources corev1.ResourceRe
 	}
 	return resources
 }
+
+func (s *ScheduleHandler) updateSchedule() error {
+	if err := s.Client.Update(s.CTX, s.schedule); err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("error updating resource %s/%s: %w", s.schedule.Namespace, s.schedule.Name, err)
+	}
+	return nil
+}
+
+func (s *ScheduleHandler) updateStatus() error {
+	err := s.Client.Status().Update(s.CTX, s.schedule)
+	if err != nil {
+		s.Log.Error(err, "Could not update SyncConfig.", "name", s.schedule)
+		return err
+	}
+	s.Log.Info("Updated SyncConfig status.")
+	return nil
+}
+
+func (s *ScheduleHandler) getOrGenerateSchedule(jobType k8upv1alpha1.JobType, originalSchedule string) string {
+	if s.schedule.Status.GeneratedSchedules == nil {
+		s.schedule.Status.GeneratedSchedules = make(map[k8upv1alpha1.JobType]string)
+	}
+	if schedule, found := s.schedule.Status.GeneratedSchedules[jobType]; found {
+		return schedule
+	}
+	newSchedule := s.randomizeSchedule(s.schedule.Namespace+"/"+s.schedule.Name+"@"+string(jobType), originalSchedule)
+	s.schedule.Status.GeneratedSchedules[jobType] = newSchedule
+	s.requireStatusUpdate = true
+	return newSchedule
+}
+
+// randomizeSchedule randomizes the given originalSchedule with a seed. The originalSchedule has to be one of the supported
+// '@x-random' predefined schedules, otherwise it returns originalSchedule unmodified.
+func (s *ScheduleHandler) randomizeSchedule(seed, originalSchedule string) (schedule string) {
+	hash := sha1.New()
```

Comment:
Using a seed helps to make sure jobs of different namespaces are distributed across time:
```
s.randomizeSchedule(s.schedule.Namespace+"/"+s.schedule.Name+"@"+string(jobType), originalSchedule)
```

Using `math.Rand` with time as a seed would mean different go routines which are running at the same time would generate the same schedule.

So I guess we could go with `math.Rand` and still provide a seed based on namespace/job name/job type 👍

Reply:
Ok, I see the need for the fixed seed. As this function is attache do `ScheduleHandler`, we can get away with just passing `jobTyp`. The namespace and name are fields already available to the function. This makes it simpler for the caller.

Reply:
There are reasons for this, please read the discussions here: https://github.com/vshn/k8up/issues/116

Reply:
`rand.NewSource(seed int64)` expects an integer as seed. Any alternatives to generating a number from a string other than Checksums? `big.Int` was chosen because checksum functions return number in hexadecimal strings. Besides, there won't be much calculations efforts: It's only calculated when a new schedule object was created, or when existing schedules have the schedules updated with a randomizer schedule but the status has not yet been updated. Once the status field is written, there will be no random calculation anymore. IMO this should be acceptable (in single cases it's not even reliably measurable)